src/Symfony/Component/Config/Definition/Builder/NodeBuilder.php

```diff
@@ -18,8 +18,8 @@
  */
 class NodeBuilder implements NodeParentInterface
 {
-    protected $parent;
-    protected $nodeMapping;
+    protected ?ParentNodeDefinitionInterface $parent = null;
```

Comment:
```suggestion
    protected (NodeDefinition&ParentNodeDefinitionInterface)|null $parent = null;
```

… in order to match the return type of `end()`. We could also think about adjusting the parameter type of `setParent()` accordingly.